### ISSUE TITLE
Undefined symbols for architecture x86_64:

### ISSUE THREAD

**Issue by minglinlu** [lukacu/mil#2]
[100%] Linking CXX executable mil
Undefined symbols for architecture x86_64:
  "cv::putText(cv::Mat&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, cv::Point_<int>, int, double, cv::Scalar_<double>, int, int, bool)", referenced from:
      Matrix<unsigned char>::drawText(char const*, float, float, int, int, int) in Matrix.cpp.o
  "trax::Metadata::Metadata(int, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)", referenced from:
      _main in BoostTrackMain.cpp.o
  "trax::Properties::get(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, char const*) const", referenced from:
      parseParameters(trax::Properties&, TrackerParams&)       in BoostTrackMain.cpp.o
  "trax::Properties::get(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, float) const", referenced from:
      parseParameters(trax::Properties&, TrackerParams&)       in BoostTrackMain.cpp.o
  "trax::Properties::get(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int) const", referenced from:
      parseParameters(trax::Properties&, TrackerParams&)       in BoostTrackMain.cpp.o
ld: symbol(s) not found for architecture x86_64
collect2: error: ld returned 1 exit status
make[2]: *** [mil] Error 1
make[1]: *** [CMakeFiles/mil.dir/all] Error 2
make: *** [all] Error 2

> I have installed traX library with opencv enabled. But I got another error, it seems like cv::putText,trax::Metadata,trax::Properties are not found when using the lib I compiled...
Everything is fine with traX when `cmake ..; make; make install` T T. I'm using open-cv-2.4.13 on mac osx 10.12.

**Comment by lukacu** [comment]
This is indeed strange, just to clarify: are you using the master branch of the TraX library? Because several things have changed since the 1.0 and I did not have time to release new version. As for the OpenCV linking error ... probably a part of OpenCV is missing or the error may go away once we resolve the other ones.

**Comment by minglinlu** [comment]
Yes, I used the master branch of the TraX library, but not sure for 1.0 version. I didn't find any version except master, java, dev branches.
However, I tested some other code, such as https://github.com/vojirt/asms which is a mean-shift tracking algorithom, no errors occured during compiling, just failed when try `traxtest -d -- main_trax`, I will probaly check it out later.
